docs/usage/shoot_high_availability_best_practices.md

```diff
@@ -0,0 +1,431 @@
+
+A few caveats though:
+
+- You cannot use HPA and VPA on the same metrics as they would influence each other, which would lead to pod trashing (more replicas require fewer resources; fewer resources require more replicas)
+- Scaling horizontally doesn't cause downtimes (at least not when out-scaling and only one replica is affected when in-scaling), but scaling vertically does (if the pod runs OOM anyway, but also when new recommendations are applied, resource requests for existing pods may be changed, which causes the pods to be rescheduled). Although the discussion is going on for a very long time now, that is still not supported in-place yet (see [KEP 1287](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/1287-in-place-update-pod-resources/README.md), [implementation in Kubernetes](https://github.com/kubernetes/kubernetes/pull/102884), [implementation in VPA](https://github.com/kubernetes/autoscaler/issues/4016)).
+
+VPA is a useful tool and Gardener-managed clusters deploy a VPA for you (HPA is supported anyway as it's built into the kube-controller-manager). You will have to create specific `VerticalPodAutoscaler` resources for your scale target and can [tweak the general VPA knobs](https://gardener.cloud/docs/gardener/api-reference/core/#verticalpodautoscaler) for Gardener-managed clusters like this:
+
+``` yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: Shoot
+spec:
+  kubernetes:
+    verticalPodAutoscaler:
+      enabled: true
+      evictAfterOOMThreshold: 10m0s
+      evictionRateBurst: 1
+      evictionRateLimit: -1
+      evictionTolerance: 0.5
+      recommendationMarginFraction: 0.15
+      updaterInterval: 1m0s
+      recommenderInterval: 1m0s
+```
+
+While horizontal pod autoscaling is relatively straight-forward, it takes a long time to master vertical pod autoscaling. We saw [performance issues](https://github.com/kubernetes/autoscaler/issues/4498), hard-coded behavior (on OOM, memory is bumped by +20% and it may take a few iterations to reach a good level), unintended pod disruptions by applying new resource requests (after 12h all targeted pods will receive new requests even though individually they would be fine without, which also drives active-passive resource consumption up), difficulties to deal with spiky workload in general (due to the algorithmic approach it takes), recommended requests may exceed node capacity, limit scaling is proportional and therefore often useless, and more. VPA is a double-edged sword: useful and necessary, but not easy to handle.
```

Comment:
It would be useful to link upstream issues (if we have any, of course) for the described drawbacks.